docs/src/main/asciidoc/amazon-lambda-http.adoc

```diff
@@ -276,3 +276,88 @@ public class MyResource {
 If you are building native images, and want to use https://aws.amazon.com/xray[AWS X-Ray Tracing] with your lambda
 you will need to include `quarkus-amazon-lambda-xray` as a dependency in your pom.  The AWS X-Ray
 library is not fully compatible with GraalVM so we had to do some integration work to make this work.
+
+== Security Integration
+
+When you invoke an HTTP request on the API Gateway, the Gateway turns that HTTP request into a JSON event document that is
+forwarded to a Quarkus Lambda.  The Quarkus Lambda parses this json and converts in into an internal representation of an HTTP
+request that can be consumed by any HTTP framework Quarkus supports (JAX-RS, servlet, Vert.x Web).
+
+API Gateway supports many different ways to securely invoke on your HTTP endpoints that are backed by Lambda and Quarkus.
+By default, Quarkus will automatically parse relevant parts of the https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html[event json document]
+and look for security based metadata and register a `java.security.Principal` internally that can be looked up in JAX-RS
+by injecting a `javax.ws.rs.core.SecurityContext`, via `HttpServletRequest.getUserPrincipal()` in servlet, and `RouteContext.user()` in Vert.x Web.
+If you want more security information, the `Principal` object can be typecast to
+a class that will give you more information.
+
+Here's how its mapped:
+
+.HTTP `quarkus-amazon-lambda-http`
+[options="header"]
+|=======================
+|Auth Type       |Principal Class                                 |Json path of Principal Name
+|Cognito JWT     |`io.quarkus.amazon.lambda.http.CognitoPrincipal`|`requestContext.authorizer.jwt.claims.cognito:username`
```

Comment:
@patriot1burke Hi Bill, I think it should implement `org.eclipse.microprofile.jwt.JsonWebToken` as we already inject JWT ID/access tokens as `JsonWebToken` with `quarkus-oidc`/`quarkus-smallrye-jwt`. It is very [simple to do](https://github.com/quarkusio/quarkus/blob/main/test-framework/security-jwt/src/main/java/io/quarkus/test/security/jwt/JwtTestSecurityIdentityAugmentorProducer.java#L32). Adding only a light-weight API dependency (it has a few interfaces only)

```
<dependency>
     <groupId>org.eclipse.microprofile.jwt</groupId>
     <artifactId>microprofile-jwt-auth-api</artifactId>
</dependency>
```

will do.
And additionally, having:
```
@RequestScoped
public class OidcJsonWebTokenProducer {
    private static final Logger LOG = Logger.getLogger(OidcJsonWebTokenProducer.class);

    @Inject
    SecurityIdentity identity;

    /**
     * The producer method for the current access token
     *
     * @return the access token
     */
    @Produces
    @RequestScoped
    JsonWebToken currentAccessToken() {
        return (JsonWebToken)identity.getPrincipal();
    }
}
```

will support the injection of Cognito JWT as `JsonWebToken`

Reply:
I just worry that AWS's interpretation of JWT will be different than microprofile's.

Reply:
@patriot1burke No, it will be fine :-). `JsonWebToken` is a representation of the actual JWT JSON payload/content - it  returns values of the claims so you can just delegate to Cognito JWT principal. 

There are some rules that complex claims  should be represented as JSON objects/etc but it is not a problem. It does not matter who issued JWT, as long as it is JWT it will be fine.

Give it a try please.

Reply:
@patriot1burke Hi Bill, it is definitely not something I'd consider a review requirement for this PR - so it is up to you if you will support `JsonWebToken`, but I'd recommend to do it - it would make Cognito tokens available for the propagation/token exchange we already support at the token propagation level (to the downstream services) and would just support the same experience users already have when dealing with JWT tokens.
Also I don't think there would be any pressure on such a would be Cogito JWT `JsonWebToken` wrapper to pass some Mp JWT TCK tests etc :-) - in most cases users only use `JsonWebToken` to access some simple claims. (`getName` - is one of `preferredUserName` or `sub` or `upn` or whatever matches in Cognito JWT case).

It is only a view into a signed token payload (no need to show the headers/do any signature verification at its level), has only 3 methods to get the claims

Reply:
@sberyozkin Not sure I want to provide the @Produces by default.  Wouldn't that conflict if the user decides not to use API Gateway security and instead uses quarkus-oidc?

Reply:
@patriot1burke That should be fine, adding an `@AlternativePriority(1)` to `OidcJsonWebTokenProducer` in `quarkus-oidc/runtime` would make sure the `quarkus-oidc` producer is preferred in such a case

Reply:
Gonna save this for another PR